core/src/main/java/org/mskcc/cbio/portal/dao/DaoTypeOfCancer.java

```diff
@@ -147,6 +154,18 @@ public static void deleteAllRecords() throws DaoException {
             JdbcUtil.closeAll(DaoTypeOfCancer.class, con, pstmt, rs);
         }
     }
+    
+    /**
+     * Deletes all records from DB using DELETE statement.
+     * 
+     * @param typeOfCancerId
+     * @throws DaoException
+     */
+    public static void deleteAllTypesOfCancer() throws DaoException {
```

Comment:
Tests calling a TRUNCATE table SQL execution, which cannot be rolled back, affect other tests depending on the order of execution of the tests. 

deleteAllTypesOfCancer() is the DELETE alternative to the deleteAllRecords() method above. I also made deleteAllRecords() deprecated so we can remove this. If we want to keep any integrity in this DB, we should not be doing TRUNCATE table anywhere in the code in my opinion. That is why I am so eager in deprecating all these methods. 

Now the real question that remains is: why would we have such "delete all" methods in the Dao layer? That is a good question... Maybe we should remove this all together an implement such tricks in the application layers that need it. I think it is also an interesting discussion to have in the context of the new API/framework @ersinciftci is working on.